File: backend/src/main/java/com/haniu/tthieu/haniu/service/PostService.java
package com.haniu.tthieu.haniu.service;

import com.haniu.tthieu.haniu.entity.marketing.Post;

import java.util.List;
import java.util.UUID;

public interface PostService {
    List<Post> getAllPosts();
    List<Post> getActivePosts();
    Post getPostById(UUID id);
    Post getPostBySlug(String slug);
    Post createPost(Post post);
    Post updatePost(UUID id, Post post);
    void deletePost(UUID id);
}

File: front-haniu/src/services/post.service.ts
import { fetchApi } from '@/lib/api';

export interface PostPayload {
  id?: string;
  title: string;
  slug?: string;
  summary?: string;
  content?: string;
  imageUrl?: string;
  active: boolean;
  publishedAt?: string;
  createdAt?: string;
  updatedAt?: string;
}

export const postService = {
  getActivePosts: async () => {
    return fetchApi('/api/v1/posts');
  },

  getAllPosts: async () => {
    return fetchApi('/api/v1/posts/all');
  },

  getPostBySlug: async (slug: string) => {
    return fetchApi(`/api/v1/posts/${slug}`);
  },

  createPost: async (payload: PostPayload) => {
    return fetchApi('/api/v1/posts', {
      method: 'POST',
      body: JSON.stringify(payload),
    });
  },

  updatePost: async (id: string, payload: PostPayload) => {
    return fetchApi(`/api/v1/posts/${id}`, {
      method: 'PUT',
      body: JSON.stringify(payload),
    });
  },

  deletePost: async (id: string) => {
    return fetchApi(`/api/v1/posts/${id}`, {
      method: 'DELETE',
    });
  }
};
